solvers.go

```diff
@@ -1160,13 +1160,13 @@ func (s *BarzilaiBorweinSolver) Step(model Nodes) error {
 					contractionAxes[axis] = axis
 				}
 
-				valGradDiffscalarProd, err := tensor.Contract(valueDiff, gradDiff, contractionAxes, contractionAxes)
+				valGradDiffscalarProd, _ := tensor.Contract(valueDiff, gradDiff, contractionAxes, contractionAxes)
```

Comment:
This is not a good idea. @siquus how should errrors be handled for Barzilai Borwein. I'm going to guess the default is to return the error?